src/templates/operations_problem_templates.jl

```diff
@@ -0,0 +1,208 @@
+"""
+function template_economic_dispatch(; kwargs...)
+
+    devices = get(
+        kwargs,
+        :devices,
+        Dict(
+            :Generators => DeviceModel(PSY.ThermalStandard, ThermalRampLimited),
+            :RE => DeviceModel(PSY.RenewableDispatch, RenewableFullDispatch),
+            :DistRE => DeviceModel(PSY.RenewableFix, RenewableFixed),
+            :Hydro => DeviceModel(PSY.HydroEnergyReservoir, HydroDispatchReservoirFlow),
+            :HydroROR => DeviceModel(PSY.HydroDispatch, HydroFixed),
+            :Loads => DeviceModel(PSY.PowerLoad, StaticPowerLoad),
+            :ILoads => DeviceModel(PSY.InterruptibleLoad, InterruptiblePowerLoad),
+        ),
+    )
+
+    services = get(kwargs, :services, Dict())
+
+    template = _generic_template(devices = devices, services = services; kwargs...)
+
+    return template
+end
+
+"""
+    EconomicDispatchProblem(system::PSY.System; kwargs...)
+
+Creates an `OperationsProblemTemplate` with default DeviceModels for an EconomicDispatch
+problem. Uses the template to create an `OperationsProblem`.
+
+# Example
+```julia
+ed_problem = EconomicDispatchProblem(system)
+```
+
+# Accepted Key Words
+- `network::Type{<:PM.AbstractPowerModel}` : override default network model settings
+- `devices::Dict{Symbol, DeviceModel}` : override default `DeviceModel` settings
+- `branches::Dict{Symbol, DeviceModel}` : override default `DeviceModel` settings
+- `services::Dict{Symbol, ServiceModel}` : override default `ServiceModel` settings
+"""
+
+function EconomicDispatchProblem(system::PSY.System; kwargs...)
+    template = template_economic_dispatch(; kwargs...)
+    op_problem = OperationsProblem(EconomicDispatchProblem, template, system)
+    return op_problem
+end
+
+"""
+    UnitCommitmentProblem(system::PSY.System; kwargs...)
+
+Creates an `OperationsProblemTemplate` with default DeviceModels for a Unit Commitment
+problem. Uses the template to create an `OperationsProblem`.
+
+# Example
+```julia
+uc_problem = UnitCommitmentProblem(system)
+```
+
+# Accepted Key Words
+- `network::Type{<:PM.AbstractPowerModel}` : override default network model settings
+- `devices::Dict{Symbol, DeviceModel}` : override default `DeviceModel` settings
+- `branches::Dict{Symbol, DeviceModel}` : override default `DeviceModel` settings
+- `services::Dict{Symbol, ServiceModel}` : override default `ServiceModel` settings
+"""
+
+function UnitCommitmentProblem(system::PSY.System; kwargs...)
+    template = template_unit_commitment(; kwargs...)
+    op_problem = OperationsProblem(UnitCommitmentProblem, template, system)
+    return op_problem
+end
+
+"""
+    run_unit_commitment(system::PSY.System; kwargs...)
+
+Creates an `OperationsProblemTemplate` with default DeviceModels for a Unit Commitment
+problem. Uses the template to create an `OperationsProblem`. Solves the created operations problem.
+
+# Example
+```julia
+results = run_unit_commitment(system; optimizer = optimizer)
+```
+
+# Accepted Key Words
```

Comment:
This is incomplete. We need to be able to pass the kwargs for the `OperationsProblem` constructor here too